app/javascript/mastodon/uuid.js

```diff
@@ -1,3 +1,3 @@
 export default function uuid(a) {
-  return a ? (a^Math.random() * 16 >> a / 4).toString(16) : ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, uuid);
+  return a ? (a^Crypto.getRandomValues() * 16 >> a / 4).toString(16) : ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, uuid);
```

Comment:
Did you test it?

* Crypto.getRandomValues() **does't returns a number**. Instead, set values to passed TypedArray.
* because getRandomValues is method of Crypto instance, so you need to use `crypto` instead of `Crypto`, which is property of globalThis (a.k.a. window, in browser).

Please read https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues once more.

(I think we seriously needs to introduce TypeScript to automatically reject invalid code by CI)

Reply:
Thanks for the information. The crypto / Crypto thing was lost on my during my read.

I'll close this, if the mastodev feel this is worth doing correctly. They can proceed as they will.

Reply:
Somebody did try to jump to TypeScript a while ago... Worth the discussion again? #16210